pyccel/ast/core.py

```diff
@@ -4593,7 +4593,13 @@ def __init__(
     def __getitem__(self, *args):
 
         if self.shape and len(self.shape) != len(args):
-            raise IndexError('Rank mismatch.')
+
+            raise IndexException('Rank mismatch.')
+
+        if len(args) == 1 and isinstance(args[0], (Tuple, tuple, list)):
```

Comment:
Should `args[0]` not be a `PythonTuple` for the syntax `a[(1,2)]`?

Reply:
this PR doesn't address that case, it's a bug in the IndexedVariable which is used in another project Psydac, but I'm not sure if the `PythonTuple` case can be addressed in the same way

Reply:
It seems I have treated this case quite clumsily in `SemanticParser` (and possibly elsewhere) by calling `__getitem__` directly. We should probably use this PR to fix that.
https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L900

Reply:
https://github.com/pyccel/pyccel/blob/master/pyccel/ast/core.py#L2813

Reply:
https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L2272
https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L2311
https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L2319

Reply:
https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L903
https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L1772
https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L1773

Reply:
I think that's all of them ^^'

Reply:
ah yes you are right we should probably use `A[i]` instead of `A.__getitem__(i)`